DigitalLearningSolutions.Web/Scripts/trackingSystem/centreCourseSetup.ts

```diff
@@ -20,20 +20,23 @@ function copyLaunchCourseLinkToClipboard(customisationId: string) {
   const launchCourseButtonId = launchCourseButtonIdPrefix + customisationId;
   const launchCourseButton = document.getElementById(launchCourseButtonId) as HTMLAnchorElement;
   const link = launchCourseButton.href;
-  const succeeded = copyTextToClipboard(link);
-  const alertMessage = succeeded
-    ? `Copied the text: ${link}`
-    : `Copy not supported or blocked. Try manually selecting and copying: ${link}`;
-
-  alert(alertMessage);
+  copyTextToClipboard(link);
 }
 
-function copyTextToClipboard(textToCopy: string): boolean {
+function copyTextToClipboard(textToCopy: string): void {
   try {
-    navigator.clipboard.writeText(textToCopy);
-    return true;
+    navigator.clipboard.writeText(textToCopy).then(() => {
+      displaySuccessAlert(textToCopy);
+    }, () => {
+      displayFailureAlert(textToCopy);

```

Comment:
Will displayFailureAlert only occur is displaySuccessAlert fails?

Reply:
displayFailureAlert only occurs if promise writeText() fails. I suppose it could be considered a little redundant here - if we removed this optional failure path, then the .then() should just throw an error in fail cases and that will get picked up by the try...catch, which would eventually display a failure.

I think it is necessary to keep the failure path on the promise in case someone has some security restrictions preventing access to the clipboard. However, I've realised there is a better way to deal with this code so we aren't relying on a try...catch. So I've refactored it to be a bit nicer.